test/13_arraycommon.jl

```diff
@@ -442,6 +461,32 @@ end
         end
     end
 
+    @testset "copyto! and copy! corner cases" begin
+        x = categorical([1:255;1:255], true)
+        y = categorical([1, 1000], true)
+        @test copyto!(x, 1, y, 1, 1)[1] == y[1]
+        @test_throws ErrorException copyto!(x, 1, y, 1, 1)[2]
```

Comment:
Better check the length? Actually, better check the whole contents of the result (same for the line above).